# Logistic Regression Model

In [23]:
# Import Dependencies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [117]:
# Importing encoded data created in Linear Regression Model
encoded_data = pd.read_csv("../data/encoded_data.csv")
encoded_data = encoded_data.drop(columns =['Platform_0', 'Platform_1',
       'Platform_2', 'Platform_3', 'Platform_4', 'Platform_5', 'Platform_6',
       'Platform_7', 'Genre_0', 'Genre_1', 'Genre_2', 'Genre_3', 'Genre_4',
       'Genre_5', 'Genre_6', 'Genre_7', 'Genre_8', 'Genre_9', 'Genre_10',
       'Genre_11',"Price","Console", "Name", "Game Title", "Platform LabelCode", "Genre LabelCode", "Publisher"])

In [118]:
encoded_data.columns

Index(['Year', 'NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales',
       'Global_Sales', 'Mean', 'Median'],
      dtype='object')

## Train, Test, Split

In [153]:
from sklearn.model_selection import train_test_split

In [154]:
X_train, X_test, y_train, y_test = train_test_split(encoded_data.drop(["Mean"],axis=1), 
                                                    encoded_data["Mean"], test_size=0.35, 
                                                    random_state=42)

## Training and Predicting

In [155]:
from sklearn.linear_model import LogisticRegression

In [156]:
logmodel = LogisticRegression()
logmodel.fit(X_train,y_train)

LogisticRegression()

In [157]:
predictions = logmodel.predict(X_test)

## Evaluation

In [163]:
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix

In [164]:
print(confusion_matrix(y_test,predictions))
print('\n')
print(classification_report(y_test,predictions))

[[295  16]
 [109  30]]


              precision    recall  f1-score   support

       False       0.73      0.95      0.83       311
        True       0.65      0.22      0.32       139

    accuracy                           0.72       450
   macro avg       0.69      0.58      0.57       450
weighted avg       0.71      0.72      0.67       450



## Summary of Analysis

* False indicates if the game price falls above the average prices for all the games in the data set. 

* Confusion Matrix:  
    In row 1, of the 311 false Mean values, 295 were predicted correctly, and 16 were predicted as true. In row 2, of the 138 true Mean values, 109 were predicted correctly and 30 were predected as false.  
  
* Classification Report:  
    The test data percentage that gave us the best F1-Score and accuracy was 35%.  
    Precision is intuitively the ability of the classifier not to label as positive a sample that is negative.  
    Recall is intuitively the ability of the classifier to find all the positive samples.  
    The F1-Score can be interpreted as a weighted harmonic mean of the precision and recall, where an F1-Score reaches its best value at 1 and worst score at 0.  
    Support is the number of occurrences of each class.